### PR TITLE
Feature/58 manage aanganwadi

### LINKED ISSUE #58: (UI) Add new Aanganwadi
**Acceptance criteria:**

1. Logged in user should be able to navigate to Add Aanganwadi screen while selecting from the menu.
2. We should see the following fields/controls like Aanganwadi person, contact phone, password, location, latitude and longitude as given the mock up.
3. When user user submits the screen should save the data to the DB and redirect to Aangawadi list screen.

**Note:**
We can have a dummy list screen with only "Add" button no need of listing the data. That will be next story.

**Reference wire mock up:**

Image

### PR DESCRIPTION
**Pull request by 786ahmedhussain**
1) Add an Aanganwadi
2) Fetch Aanganwadi List
3) Delete an Aanganwadi
4) Edit Current Aanganwandi - (Need to address this data binding)

### CODE REVIEW

File: FrontEnd/src/router/index.js [hunk 1]
```diff
@@ -46,6 +46,21 @@ const router = createRouter({
       name: "ChildSupplementarySummaryView",
       component: () => import("../views/ChildSupplementarySummaryView.vue"),
     },
+    {
+      path: "/manageAanganwadi",
```
Comment (mjayakumarmca) [inline FrontEnd/src/router/index.js]: Can we rename this file as "AddAanganwadi" instead of Manage

File: FrontEnd/src/stores/aanganwadi.js [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+  }),
+  getters: {
+    getAanganwadies(state) {
+      return state.aanganwadies;
+    },
+  },
+  actions: {
+    async fetchAanganwadies() {
+      try {
+        const data = await axios.get("http://127.0.0.1:7000/api/getAanganwadis");
```
Comment (mjayakumarmca) [inline FrontEnd/src/stores/aanganwadi.js]: We don't need to provide full url just the endpoint is enough. you can refer NGO store for this. All the below endpoints can be also corrected like that.

File: FrontEnd/src/views/AanganwadiList.vue [hunk 3]
```diff
@@ -0,0 +1,131 @@
+<template>
+    <div class="row">
+      <div class="col-12 p-2">
+        <h3 class="float-start ps-2">All Aanganwadies</h3>
+        <router-link to="/manageAanganwadi" custom v-slot="{ navigate }">
```
Comment (mjayakumarmca) [inline FrontEnd/src/views/AanganwadiList.vue]: Here too rename it.

File: FrontEnd/src/views/ManageAanganwadi.vue [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+              />
+            </div>
+            
+            
+            <div class="row">
+              <div class="col-12 p-2">
+                <button
+                  type="submit"
+                  class="btn btn-primary float-end mx-2"
+                  data-nav="admin.ngos"
```
Comment (mjayakumarmca) [inline FrontEnd/src/views/ManageAanganwadi.vue]: Change the data-nav value of admin.ngos to admin.aaganwadis here in other places too

File: FrontEnd/src/views/ManageAanganwadi.vue [hunk 5]
```diff
@@ -0,0 +1,128 @@
+<template>
+    <div class="container-md mt-5 p-3">
+      <div id="x-contest" class="container-float">
+        <div class="row">
+          <h3 class="float-start">Manage Aanganwadi</h3>
+          <form @submit.prevent="postAanganwadi">
```
Comment (mjayakumarmca) [inline FrontEnd/src/views/ManageAanganwadi.vue]: Fix the UI alignment like NGO here

File: FrontEnd/src/views/ManageAanganwadi.vue [hunk 6]
```diff
@@ -0,0 +1,128 @@
...
+              <input
+                type="text"
+                class="form-control"
+                id="exampleFormControlInput1"
+                placeholder="Latitude, Longitude Coordinates"
+                v-model="aanganwadi.coordinates"
+              />
+            </div>
+            
+
```
Comment (mjayakumarmca) [inline FrontEnd/src/views/ManageAanganwadi.vue]: remove extra line space

File: FrontEnd/src/views/ManageAanganwadi.vue [hunk 7]
```diff
@@ -0,0 +1,128 @@
...
+    }
+  }
+  </style>
+  
+  <script setup>
+  import { ref, onMounted, computed, reactive } from "vue";
+  import { useAanganwadiStore } from "../stores/aanganwadi";
+  import router from "../router";
+  let aanganwadi = reactive({
+    aanganwadiName: "Test",
```
Comment (mjayakumarmca) [inline FrontEnd/src/views/ManageAanganwadi.vue]: Remove Test make it empty

File: FrontEnd/src/views/ManageAanganwadi.vue [hunk 8]
```diff
@@ -0,0 +1,128 @@
...
+    contactPersonName: "",
+    contactPersonPhone: 0,
+    contactPersonEmail:"dummy@hotmail.com",
```
Comment (mjayakumarmca) [inline FrontEnd/src/views/ManageAanganwadi.vue]: can we remove this also

File: FrontEnd/src/views/ManageAanganwadi.vue [hunk 9]
```diff
@@ -0,0 +1,128 @@
...
+    contactPersonPassword: "",
+    location: "",
+    //coordinates: "",
+    taluka:"Test Taluka",
```
Comment (mjayakumarmca) [inline FrontEnd/src/views/ManageAanganwadi.vue]: here too and pincode too

File: FrontEnd/src/views/EditAanganwadi.vue [hunk 10]
```diff
@@ -0,0 +1,138 @@
...
+              />
+            </div>
+            
+            
+            <div class="row">
+              <div class="col-12 p-2">
+                <button
+                  type="submit"
+                  class="btn btn-primary float-end mx-2"
+                  data-nav="admin.ngos"
```
Comment (mjayakumarmca) [inline FrontEnd/src/views/EditAanganwadi.vue]: rename all ngos to aaganwadis

File: FrontEnd/src/views/AanganwadiList.vue [hunk 11]
```diff
@@ -0,0 +1,131 @@
...
+  };  
+
+  const editAanganwadiPage = (id) => {
+    router.replace({ path: '/editAanganwadiPage/'+id })
+  };
+
+  onMounted(() => {
+    store.fetchAanganwadies();
+  });
+const editUser = (item)=> {
```
Comment (mjayakumarmca) [inline FrontEnd/src/views/AanganwadiList.vue]: have proper alignment here

File: FrontEnd/src/views/AanganwadiList.vue [hunk 12]
```diff
@@ -0,0 +1,131 @@
...
+        swal("Aanganwadi has been deleted!", {
+          icon: "success",
+        });
+        store.fetchAanganwadies();
+      }
+    });
+  };
+
+  const editAanganwadi = (id) => {
+    swal({
```
Comment (mjayakumarmca) [inline FrontEnd/src/views/AanganwadiList.vue]: For edit we don't need confirmation dialog logic just direct edit is fine. so we can remove this here.

File: FrontEnd/src/views/AanganwadiList.vue [hunk 13]
```diff
@@ -0,0 +1,131 @@
...
+            </td>
+            
+          </tr>
+        </tbody>
+      </table>
+    </div>
+  </template>
+
+
+
```
Comment (mjayakumarmca) [inline FrontEnd/src/views/AanganwadiList.vue]: remove extra line spaces

**Review by mjayakumarmca — CHANGES_REQUESTED** [review]
Given few comments to address

File: FrontEnd/src/router/index.js (hunk 1 above)
Comment (786ahmedhussain) [reply FrontEnd/src/router/index.js]: end point renamed to addAanganwadi and vue page renamed to AddAanganwadi

File: FrontEnd/src/views/AanganwadiList.vue (hunk 3 above)
Comment (786ahmedhussain) [reply FrontEnd/src/views/AanganwadiList.vue]: endpoint renamed to addAanganwadi

File: FrontEnd/src/views/ManageAanganwadi.vue (hunk 4 above)
Comment (786ahmedhussain) [reply FrontEnd/src/views/ManageAanganwadi.vue]: data-nav value renamed to admin.aanganwadis

File: FrontEnd/src/views/EditAanganwadi.vue (hunk 10 above)
Comment (786ahmedhussain) [reply FrontEnd/src/views/EditAanganwadi.vue]: data-nav value renamed to admin.aanganwadis

File: FrontEnd/src/views/AanganwadiList.vue (hunk 13 above)
Comment (786ahmedhussain) [reply FrontEnd/src/views/AanganwadiList.vue]: Extra Line Removed

**Review by mjayakumarmca — COMMENTED** [review]
LGTM

**Review by mjayakumarmca — APPROVED** [review]
LGTM

**Pull request merged**